Recruit-Training/Advanced-Recruit-Training/Embedded-Teaser/Training/Core/Src/Training.c

```diff
@@ -1,6 +1,60 @@
 #include "Training.h"
+/*
+    CONDITIONS FOR FAILURE
+    
+    -the velocity and direction bit don't express the same direction
+    -the input for the first and second motor aren't EXACTLY the same
+    -more than one headlight choice is chosen
+    -both left and right signals are on at the same time
 
+*/
 void trainingTask(uint8_t* data)
 {
+    uint8_t tempVelocityMotor1 = data[0] >> 6;
+    tempVelocityMotor1 = tempVelocityMotor1 << 7;
 
+    uint8_t tempVelocityMotor2 = data[0] >> 6;
+    tempVelocityMotor2 = tempVelocityMotor2 << 7;
+
+    if ((data[0] <<7 != tempVelocityMotor1)
+         && (data[1] << 7 != tempVelocityMotor2))
+    {
+        if (data[0] == data[1])
```

Comment:
The speeds can be different. E.g. turning. So doing a direct comparison isn't fully correct

Reply:
However, when the motors are off, this is correct as the speed of both will be 0. So just modify this a lil bit